jwst/resample/gwcs_drizzle.py

```diff
@@ -55,70 +57,61 @@ def __init__(self, product, outwcs=None, wt_scl=None,
             The value a pixel is set to in the output if the input image does
             not overlap it. The default value of NAN sets NaN values.
         """
-
         # Initialize the object fields
         self._product = product
-        self.outsci = None
-        self.outwht = None
-        self.outcon = None
-        self.uniqid = 0
 
-        if wt_scl is None:
-            self.wt_scl = ""
-        else:
-            self.wt_scl = wt_scl
-        self.kernel = kernel
-        self.fillval = fillval
         self.pixfrac = pixfrac
 
-        self.sciext = "SCI"
-        self.whtext = "WHT"
-        self.conext = "CON"
-
-        out_units = "cps"
-
         self.outexptime = product.meta.exposure.measurement_time or 0.0
 
         self.outsci = product.data
+        self.outwht = product.wht
+
         if outwcs:
             self.outwcs = outwcs
         else:
             self.outwcs = product.meta.wcs
 
-        self.outwht = product.wht
-        self.outcon = product.con
-
-        if self.outcon.ndim == 2:
-            self.outcon = np.reshape(self.outcon, (1,
-                                                   self.outcon.shape[0],
-                                                   self.outcon.shape[1]))
-        elif self.outcon.ndim != 3:
-            raise ValueError("Drizzle context image has wrong dimensions: \
-                {0}".format(product))
-
         # Check field values
         if not self.outwcs:
             raise ValueError("Either an existing file or wcs must be supplied")
 
-        if out_units == "counts":
-            np.divide(self.outsci, self.outexptime, self.outsci)
-        elif out_units != "cps":
-            raise ValueError("Illegal value for out_units: %s" % out_units)
+        ctx = product.con
+        begin_ctx_id = 0
+        if ctx is not None:
+            if ctx.size == 0:
+                ctx = None
+            elif ctx.ndim not in [2, 3]:
+                # TODO: this message seems odd
+                raise ValueError(
+                    f"Drizzle context image has wrong dimensions: {product}"
+                )
+
+        super().__init__(
+            kernel=kernel,
+            fillval=fillval,
+            out_shape=None,
+            out_img=product.data,
+            out_wht=product.wht,
+            out_ctx=ctx,
+            exptime=self.outexptime,
+            begin_ctx_id=begin_ctx_id,
+            max_ctx_id=n_max_images,
+            disable_ctx=disable_ctx
+        )
+
+        # Since the context array is dynamic, it must be re-assigned
+        # back to the product's `con` attribute.
+        product.con = self.out_ctx
```

Comment:
in my tests it worked - it simply creates an zeroes array of the same shape as the data array.

Reply:
Ok, but that means that the context array is still pre-allocated and takes up memory even though it doesn't do anything.

Also, while this may work as you intended, I would argue that it's unclear, and it would be easy to replace it with something like `model.con = np.zeros_like(model.con)`

Reply:
I don't think we need to replace anything if context is `None`, i.e., I'll have `if con is not None then assign`